src/core/utils/BirthDeathForwardSimulator.cpp

```diff
@@ -32,20 +34,42 @@ bool BirthDeathForwardSimulator::checkParameters( void ) const
         throw RbException("You need to provide the same number of time intervals as speciation rates.");
     }
 
-//    if ( mu.size() != NUM_TIME_INTERVALS )
-//    {
-//        throw RbException("You need to provide the same number of time intervals as extinction rates.");
-//    }
-//
-//    if ( phi.size() != NUM_TIME_INTERVALS )
-//    {
-//        throw RbException("You need to provide the same number of time intervals as fossilization rates.");
-//    }
-//
-//    if ( r.size() != NUM_TIME_INTERVALS )
-//    {
-//        throw RbException("You need to provide the same number of time intervals as treatment probabilities.");
-//    }
+   if ( !(mu.size() == NUM_TIME_INTERVALS || mu.size() == 1) )
+   {
+       throw RbException("You need to provide the same number of time intervals as extinction rates.");
+   }
+
+   if ( !(phi.size() == NUM_TIME_INTERVALS || phi.size() == 1) )
+   {
+       throw RbException("You need to provide the same number of time intervals as fossilization rates.");
+   }
+
+   if ( !(r.size() == NUM_TIME_INTERVALS || r.size() == 1) )
+   {
+       throw RbException("You need to provide the same number of time intervals as treatment probabilities.");
+   }
+
+   if ( !(Lambda.size() == NUM_TIME_INTERVALS || Lambda.size() == 1) )
+   {
+       throw RbException("You need to provide the same number of time intervals as speciation probabilities.");
+   }
+
+   if ( !(Mu.size() == NUM_TIME_INTERVALS || Mu.size() == 1) )
+   {
+       throw RbException("You need to provide the same number of time intervals as extinction probabilities.");
+   }
+
+   if ( !(Phi.size() == NUM_TIME_INTERVALS || Phi.size() == 1) )
+   {
+       throw RbException("You need to provide the same number of time intervals as fossilization probabilities.");
+   }
+
+   if ( !(R.size() == NUM_TIME_INTERVALS || R.size() == 1) )
+   {
+       std::cout << "R.size() == " << R.size() << std::endl;
```

Comment:
These debug messages should be in RBOUT or deleted.

Reply:
Sure, I removed this debug output in commit `5f590d1b`.